### PR TITLE
Spelling check

### LINKED ISSUE #228: Consider using check-spelling action
We seem to have a tendency to introduce spelling errors. We might consider adding https://github.com/marketplace/actions/check-spelling

### PR DESCRIPTION
**Pull request by admiralAwkbar**
This adds the check spelling actions to avoid some typos. Fixes #228 

Note:
This is still WIP, as I would like early feedback on the way I'm handling the expected words, and all the tweaking needed.
Also, ping @jsoref as you did the initial PR #220 fixing some things (and are the creator of the action), I'm not sure the config you used. Fell free to leverage my branch to add up any work you may already do on this.

Update: This is now ready for review

This closes #235

### CODE REVIEW

**Comment by admiralAwkbar** [comment]
@GaboFDC seems we have a permission error:
```
Preparing a comment
{
  "body": "\n#### New misspellings found, please review:\n\n* cdxml\n* changelog\n* chmod\n* Dockerhub\n* EOL\n* GPR\n* IAm\n* icu\n* jre\n* kotlin\n* krb\n* ktlint\n* libgcc\n* libintl\n* libssl\n* libstdc\n* linux\n* lttng\n* microsoft\n* ncurses\n* nq\n* openjdk\n* php\n* powershell\n* println\n* psd\n* psm\n* psrc\n* pssc\n* psscriptanalyzer\n* pwsh\n* rcu\n* rhc\n* ry\n* terminfo\n* tzdata\n* userspace\n* xargs\n* zlib\n\n<details><summary>To accept these changes, run the following commands</summary>\n\n\n```\nperl -e '\nmy $new_expect_file=\".github/actions/spelling/expect.txt\";\nopen FILE, q{<}, $new_expect_file; chomp(my @words = <FILE>); close FILE;\nmy @add=qw('\"cdxml changelog chmod Dockerhub EOL GPR IAm icu jre kotlin krb ktlint libgcc libintl libssl libstdc linux lttng microsoft ncurses nq openjdk php powershell println psd psm psrc pssc psscriptanalyzer pwsh rcu rhc ry terminfo tzdata userspace xargs zlib \"');\nmy %items; @items{@words} = @words x (1); @items{@add} = @add x (1);\n@words = sort {lc($a) cmp lc($b)} keys %items;\nopen FILE, q{>}, $new_expect_file; for my $word (@words) { print FILE \"$word\\n\" if $word =~ /\\w/; };\nclose FILE;'\ngit add .github/actions/spelling || echo '... you want to ensure .github/actions/spelling/expect.txt is added to your repository...'\n```\n</details>\n"
}
https://api.github.com/repos/github/super-linter/commits/d6b1e5121299981ccf8b1177aa0b8202b66a1858/comments
{
  "message": "Bad credentials",
  "documentation_url": "https://developer.github.com/v3"
}
```

File: .github/workflows/check-spelling.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+      #############################
+      # Run check spelling action #
+      #############################
+      - name: Check spelling
+        uses: check-spelling/check-spelling@0.0.16-alpha
+        with:
+          bucket: .github/actions
+          project: spelling
+
```
Comment (jsoref) [inline .github/workflows/check-spelling.yml:47]:
```suggestion
          project: spelling
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

Comment (jsoref) [reply .github/workflows/check-spelling.yml:47]: I think this will fix the permissions

**Comment by zkoppert** [comment]
Our strategy with this spell check action is going to be to merge this in and monitor if the tool provides enough helpful results given the cost of regularly updating the `.github/actions/spelling/expect.txt`.

**Comment by jsoref** [comment]
I'm working on making it possible to talk to a bot to have it perform the updates to the file.

Sadly I'm missing access to a certain hook (comment) which means it's moderately painful to implement.

File: .github/workflows/check-spelling.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# Documentation:
+# https://help.github.com/en/articles/workflow-syntax-for-github-actions
+#
+
+name: Spell checking
+#############################
+# Start the job on all push #
+#############################
+on:
+  push:
```
Comment (jsoref) [inline .github/workflows/check-spelling.yml:16]: Oh, I should note that one can use `push` or `pull_request`.

I've personally favored the former, but if someone forks a repository and doesn't enable GitHub actions, then it won't run. So I'm slowly rethinking this preference.

On my todo list is teaching the bot to recognize when there's an older comment by it and have it obsolete it. I _think_ that'd allow one to use both `push` and `pull_request` w/o worrying too much.

**Pull request merged**